privacyidea/lib/caconnectors/msca.py

```diff
@@ -20,52 +20,62 @@
 This implementation is for the Microsoft CA via our middleware.
 
 This module is tested in tests/test_lib_caconnector.py
-
-TODO: write tests
 """
 
 from privacyidea.lib.error import CAError
 from privacyidea.lib.caconnectors.baseca import BaseCAConnector, AvailableCAConnectors
 from OpenSSL import crypto
 import logging
+import traceback
 from six.moves import input
 from privacyidea.lib.utils import is_true
 from privacyidea.lib.error import CSRError, CSRPending
+from OpenSSL.crypto import dump_privatekey, load_privatekey
```

Comment:
We should use [`pyca/cryptography`](https://cryptography.io/en/latest/) for x509 certificate handling. The `PyOpenSSL` projects states that it should only be used for SSL/TLS handling: https://pypi.org/project/pyOpenSSL/

Reply:
OK, I think I got it
Done in 85bc05c6